crates/util/python-utils/src/bridge/drt.rs

```diff
@@ -115,13 +130,39 @@ fn deposit_request_transaction_inner(
         // NOTE: the deposit won't be found by the sequencer if the order isn't correct.
         builder.ordering(TxOrdering::Untouched);
         builder.add_recipient(bridge_in_address.script_pubkey(), BRIDGE_IN_AMOUNT);
-        builder.add_data(&op_return_data);
+
+        builder.add_data(&PushBytesBuf::try_from(op_return_data).expect("not a valid push bytes"));
+
         builder.fee_rate(fee_rate);
         builder.finish().expect("drt: invalid psbt")
     };
     wallet.sign(&mut psbt, Default::default()).unwrap();
 
     let tx = psbt.extract_tx().expect("drt: invalid tx");
+
+    // Create DepositRequestData and add it to the static vector
+    {
+        let mut storage = DEPOSIT_REQUEST_DATA_STORAGE.lock().unwrap();
+        let stake_index = storage.len() as u32; // Use the position in vector as stake index
```

Comment:
This should be implemented by creating a type (exporting it to Python) which has this state, and then these functions could be functions on that type to manipulate it.

Or really, why does this even need to track this data?  I don't see it being used anywhere else.  We could just pass in the index as an argument.

Using a global like this will break, since it persists across test invocations, so when we spin down a test env and spin up the next env, tests running in that env will break because they'll be using the old stake chain index.